client/state/reader/lists/reducer.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import keyBy from 'lodash/keyBy';
+import map from 'lodash/map';
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_LISTS_RECEIVE,
+	READER_LISTS_REQUEST,
+	READER_LISTS_REQUEST_SUCCESS,
+	READER_LISTS_REQUEST_FAILURE,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+import { itemsSchema, subscriptionsSchema } from './schema';
+import { isValidStateWithSchema } from 'state/utils';
+
+/**
+ * Tracks all known list objects, indexed by list ID.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = {}, action ) {
+	switch ( action.type ) {
+		case READER_LISTS_RECEIVE:
+			return Object.assign( {}, state, keyBy( action.lists, 'ID' ) );
+		case SERIALIZE:
+			return state;
+		case DESERIALIZE:
+			if ( ! isValidStateWithSchema( state, itemsSchema ) ) {
+				return {};
+			}
+			return state;
+	}
+	return state;
+}
+
+/**
+ * Tracks which list IDs the current user is subscribed to.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function subscribedLists( state = {}, action ) {
+	switch ( action.type ) {
+		case READER_LISTS_RECEIVE:
+			return Object.assign( {}, state, map( action.lists, 'ID' ) );
```

Comment:
As it stands, you're using the array index from the map to hold each ID. that means you could have the same ID in the map more than once. expanded, it's `assign( {}, { 0: 1234 }, { 0: 345, 1: 4214, 2: 4444 } )`

So it would even overwrite any existing keys...